R/tax_range_strat.R

```diff
@@ -0,0 +1,225 @@
+#'   (first appearance; default), "LAD" (last appearance), or "name"
+#'   (alphabetically by taxon names).
+#' @param plot_args A list of optional arguments that are passed directly to
+#'   [graphics::plot()]. Useful arguments include `xlab` (the x-axis label),
+#'   `ylab` (the y-axis label, default is "Bed number"), `main` (the plot
+#'   title), `xlim` (the x-axis limits), and `ylim` (the y-axis limits). The
+#'   `axes` and `type` arguments are not supported and will be overridden.
+#' @param x_args A list of optional arguments that are passed directly to
+#'   [axis()] when generating the x-axis. Useful arguments include `font` (e.g.,
+#'   `3` is italic) and `las` (label orientation).
+#' @param y_args A list of optional arguments that are passed directly to
+#'   [axis()] when generating the y-axis. Useful arguments include `font` (e.g.,
+#'   `3` is italic) and `las` (label orientation).
+#'
+#' @return Invisibly returns a data.frame of the calculated taxonomic
+#'   stratigraphic ranges.
+#'
+#'   The function is usually used for its side effect, which is to create a plot
+#'   showing the stratigraphic ranges of taxa in a section, with levels at which
+#'   the taxon was sampled indicated with a point.
+#'
+#' @section Developer(s): Bethany Allen, William Gearty & Alexander Dunhill
+#' @section Reviewer(s): William Gearty & Lewis A. Jones
+#' @importFrom graphics axis par segments plot points box
+#'
+#' @examples
+#' #Load tetrapod dataset
+#' data(tetrapods)
+#' # Sample tetrapod occurrences
+#' tetrapod_names <- tetrapods$accepted_name[1:50]
+#' # Simulate bed numbers
+#' beds_sampled <- sample.int(n = 10, size = 50, replace = TRUE)
+#' # Simulate certainty values
+#' certainty_sampled <- sample(x = 0:1, size = 50, replace = TRUE)
+#' #Combine into data frame
+#' occdf <- data.frame(taxon = tetrapod_names,
+#'                     bed = beds_sampled,
+#'                     certainty = certainty_sampled)
+#' # Plot stratigraphic ranges
+#' par(mar = c(12, 5, 2, 2))
+#' tax_range_strat(occdf, name = "taxon")
+#' tax_range_strat(occdf, name = "taxon", certainty = "certainty",
+#'                 plot_args = list(ylab = "Stratigraphic height (m)"))
+#' # Plot stratigraphic ranges with more labelling
+#' tax_range_strat(occdf, name = "taxon", certainty = "certainty", by = "name",
+#'                 plot_args = list(main = "Section A",
+#'                                  ylab = "Stratigraphic height (m)"))
+#' eras_custom <- data.frame(name = c("Mesozoic", "Cenozoic"),
+#'                           max_age = c(0.5, 3.5),
+#'                           min_age = c(3.5, 10.5),
+#'                           color = c("#67C5CA", "#F2F91D"))
+#' axis_geo(side = 4, intervals = eras_custom, tick_labels = FALSE)
+#'
+#' @export
+tax_range_strat <- function(occdf, name = "genus", level = "bed",
+                            certainty = NULL, by = "FAD",
+                            plot_args = list(xlab = "", ylab = "Bed number"),
```

Comment:
Something to be aware of is that you can't really use xlab currently as taxonomic names are plotted on top of it. I don't think this really matters here because it is rare(?) to see an xlab/title on range charts? However something you might want to consider updating in the future.

Reply:
Surely you could use it if you really wanted to, you would just need to be careful about the placement?

Reply:
I think it is a pain within the plot function to modify the location of the x-axis title. However, you could update the function by always adding the xlab after you've added the x-axis labels, e.g.:

```title(xlab = "Taxa", line = 4)```

Reply:
Yeah, it doesn't look very straightforward to modify the x-axis title via `plot()` (the `mgp` parameter seems closest, but it affects all of the titles, not just the x-axis title), and `axis()` doesn't have the option to plot the axis title, which is frustrating. If we call `title()` within the function, should we have yet another argument for the list of arguments that should be passed to that (e.g., title_args)? Another option would be to just show in the example how a user might use the `title()` function after calling `tax_range_strat()`?

Reply:
I think the most straight forward thing would be just to use `title()` and calculate a value for line argument based on the  `nchar()` of the longest species names on x? Unless there is a way to extract the exact coordinates of plotted labels? I don't think we necessarily need to make all arguments available from `title()` as we are just treating it as our "xlab". However, given that most users might not even want an xlab, I think your suggestion of showing how one might use `title()` is probably OK too?

Reply:
I don't know much about base R and getting the coordinates of plotted labels, but it sounds complicated. I'm in favor of just including an example showing how to use `title()` (and maybe adding some text in the `x_args` documentation that says that users might want to use `title()` instead of supplying `xlab`).